TypedSyntax/test/runtests.jl

```diff
@@ -1,9 +1,11 @@
 using JuliaSyntax: JuliaSyntax, SyntaxNode, children, child, sourcetext, kind, @K_str
 using TypedSyntax: TypedSyntax, TypedSyntaxNode
 using Dates, InteractiveUtils, Test
+using Core.Compiler: widenconst
 
-has_name_typ(node, name::Symbol, @nospecialize(Ts::Tuple)) = kind(node) == K"Identifier" && node.val === name && node.typ in Ts
-has_name_typ(node, name::Symbol, @nospecialize(T)) = kind(node) == K"Identifier" && node.val === name && node.typ === T
+has_typ(node, @nospecialize(Ts::Tuple)) = any(has_typ(node, T) for T in Ts)
+has_typ(node, @nospecialize(T)) = node.typ === T || !isnothing(node.typ) && widenconst(node.typ) === T
+has_name_typ(node, name::Symbol, @nospecialize(T)) = kind(node) == K"Identifier" && node.val === name && has_typ(node, T)
```

Comment:
Not 100% sure about these changes. Essentially, what I have been assuming is that it doesn't matter whether the node type is a `Const` or not, so long as the type of the value is correct, if we test it against a non-`Const` type.

Reply:
Yeah, this change looks reasonable.